### PR TITLE
Added travis + tests + cs fixes

### PR DESCRIPTION
**Pull request by mickaelandrieu**
This contribution is about adding premises of a test suite and coding styles.

Also, a configuration for Travis.

### CODE REVIEW

File: controllers/index.php [hunk 1]
```diff
@@ -19,7 +19,7 @@
 * needs please refer to http://www.prestashop.com for more information.
 *
 *  @author PrestaShop SA <contact@prestashop.com>
-*  @copyright  2007-2016 PrestaShop SA
+*  @copyright  2007-2016 PrestaShop SA
```
Comment (PierreRambaud) [inline controllers/index.php]: 2016?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+before_script:
+    - ./tools/move_module.sh
+    - $MODULE_SCRIPTS_DIR/get_prestashop.sh
+    - $MODULE_SCRIPTS_DIR/install_webserver.sh
+    - $MODULE_SCRIPTS_DIR/start_prestashop.sh
+    - $MODULE_SCRIPTS_DIR/install_module.sh gamification
+    
+script: 
+    - cd $MODULE_DIR
```
Comment (Quetzacoalt91) [inline .travis.yml]: It seems this line is unused.

```suggestion

```

File: controllers/index.php (hunk 1 above)
Comment (mickaelandrieu) [reply controllers/index.php]: if I update this one I need to update all the others, I'd like to keep this pull request with one scope *only*

File: composer.json [hunk 3]
```diff
@@ -10,10 +10,22 @@
...
+    "symfony/phpunit-bridge": "^3.4",
+    "symfony/debug": "^3.4"
   },
   "config": {
     "preferred-install": "dist"
   },
+  "scripts": {
+    "cs-fix": "@php ./vendor/bin/php-cs-fixer fix .",
+    "cs-fix-test": "@php ./vendor/bin/php-cs-fixer fix --dry-run --stop-on-violation --show-progress=dot .",
+    "test": "SYMFONY_PHPUNIT_VERSION=5.7 php ./vendor/bin/simple-phpunit"
```
Comment (mickaelandrieu) [inline composer.json]: interesting point here: passing env variables here make the special @php unavailable in Composer scripts. I don't know if it's intended or not.

**Comment by matks** [comment]
@mickaelandrieu Wen being shipped to Addons as a downloadable PS module, do you know what process is performed ? I think there is at least 1 `composer install`.
- is the composer install performed for DEV or PROD
- shall we remove the test files (`phpunit.xml`, `.travis.yml`) and folders (`tests` and `tools`) as they are not useful for the module to be executed/installed ?

**Comment by matks** [comment]
⚠️ this PR targets `master` instead of `dev`, is that intended ?

**Comment by mickaelandrieu** [comment]
> is the composer install performed for DEV or PROD?

I don't think composer is called at all, this is why I don't use it to autoload module classes

> shall we remove the test files (phpunit.xml, .travis.yml) and folders (tests and tools) as they are not useful for the module to be executed/installed ?

We could, but how? I'm thinking about using `deploy` feature of traverse to perform these tasks, but this wont be addressed here.

> this PR targets master instead of dev, is that intended ?

I don't know because I can't figure out why we have both `dev` and `master` branches on every module :/ ping @Quetzacoalt91

**Comment by Quetzacoalt91** [comment]
* Composer is run with --no-dev parameter when deployed to the marketplace. This allow the zip to not provide useless stuff for the merchants.

* `dev` contains the current version in progress. Once we are ready to publish a new version, we merge `merge` into `master` to trigger a new release on the marketplace the next night.

**Comment by mickaelandrieu** [comment]
I've switched to `dev` and as I have only dev dependencies, there won't be any impact for the marketplace.

Thanks for the answers @Quetzacoalt91 !

**Comment by matks** [comment]
Thanks @mickaelandrieu

**Pull request merged**